src/Unit-2/lesson6/README.md

```diff
@@ -0,0 +1,299 @@
+                };
+
+                store.Initialize();
+
+                return store;
+            });
+
+        public static IDocumentStore Store =>
+            LazyStore.Value;
+    }
+}
+````
+
+Remember to have started the RavenDB server.
+
+### Step 4: Loading the company information
+
+Now, it's time to load a company document from the server.
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        using (var session = DocumentStoreHolder.Store.OpenSession())
+        {
+            var company = session.Load<Company>("companies/89");
+            Console.WriteLine(company.Name);
+        }
+    }
+}
+
+Easy. Right?
+
+There is nothing wrong here. But looking closer what happened you will see that
+the client API sent the following request to the server.
+
+````
+http://localhost:8080/databases/Northwind/docs?id=companies%2F89
+````
+
+Which produced the following response:
+
+````json
+{
+    ExternalId: "WHITC",
+    Name: "White Clover Markets",
+    Contact: {
+        Name: "Karl Jablonski",
+        Title: "Owner"
+    },
+    Address: {
+        Line1: "305 - 14th Ave. S. Suite 3B",
+        Line2: null,
+        City: "Seattle",
+        Region: "WA",
+        PostalCode: "98128",
+        Country: "USA"
+    },
+    Phone: "(206) 555-4112",
+    Fax: "(206) 555-4115"
+}
+````
+
+A lot of information if you just need the `Name` property.
+
+> It's important to understand that RavenDB is smart enough to correctly
+load data from response in a "small" model class. But, it's not exactly good
+to receive all this data if you don't need it.
+
+### Step 5: Writing your first transformer
+
+Transformers are LINQ-based server side projection functions. Just it! 
+
+Similarly to indexes, you create transformers definitions writing a class.
+
+````csharp
+public class Company_JustName : AbstractTransformerCreationTask<Company>
+{
+    public Company_JustName()
+    {
+        TransformResults = companies =>
+            from company in companies
+            select new {company.Name};
+    }
+}
+````
+
+Pretty awesome! 
+
+### Step 6: Using transformers
+
+Now you wrote your first transformer, it's time to use it.
+
+````csharp
+static void Main(string[] args)
+{
+    new Company_JustName().Execute(DocumentStoreHolder.Store);
+    using (var session = DocumentStoreHolder.Store.OpenSession())
+    {
+        var company = session.Load<Company_JustName, Company>("companies/89");
+        Console.WriteLine(company.Name);
+    }
+}
+````
+
+Because transformers are server side artifacts, you need to create tham on the
+server before thay can be used. You can do it calling the `Execute` method (as in the 
+sample code) or using `IndexCreation.CreateIndexes` function.
+
+Now, looking closer what happened you will see that the client API sent the following request to the server:
+
+````
+http://localhost:8080/databases/Northwind/queries/?&transformer=JustCompanyName&id=companies%2F89
+````
```

Comment:
Pretty sure transformer's name will be `Company/JustName` or `Company%2FJustName`